libraries/botbuilder-core/src/activityHandler.ts

```diff
@@ -443,6 +449,23 @@ export class ActivityHandler extends ActivityHandlerBase {
         throw new Error('NotImplemented');
     }
 
+    /*
+     * Handle _healthCheck invoke activity type_.
+     * 
+     * @param context The context object for the current turn.
+     * 
+     * @remarks
+     * Overwrite this method to customize or extended the built in healthCheck behavior.
+    */
+    protected async onHealthCheck(context: TurnContext): Promise<HealthCheckResponse> {
+        const  adapter = <any>context.adapter;
+        if (adapter.healthCheck && typeof adapter.healthCheck === 'function') {
```

Comment:
>adapter.healthCheck && [](start = 12, length = 22)

you can just do the "typeof" check... You don't need the first check

Reply:
OK I'll update that. You guys are cool with this runtime type identification - that's good! you see why I had to do it this way - because of the botbuilder/botbuilder-core split.